### ISSUE TITLE
Incomplete support (False negatives) on EQ_COMPARING_CLASS_NAMES

### ISSUE THREAD

**Issue by gloNelson** [spotbugs/spotbugs#1496]
Hi, I found that SpotBugs has limited support on **EQ_COMPARING_CLASS_NAMES**.

The **EQ_COMPARING_CLASS_NAMES** in [bug descriptions page](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#eq-equals-method-compares-class-names-rather-than-class-objects-eq-comparing-class-names) mentioned:
> This method checks to see if two objects are the same class by checking to see if the names of their classes are equal. You can have different classes with the same name if they are loaded by different class loaders. Just check to see if the class objects are the same.

As it can be seen from the description, it is supposed to detect the pattern likes `obj.getClass().getName().equals(obj1.getClass().getName())` 
Therefore, I wrote a test case and the SpotBugs did not report ** EQ_COMPARING_CLASS_NAMES **.
```java
public boolean test(final Object obj) {
    return obj.getClass().getName().equals(obj.getClass().getName()); // no EQ_COMPARING_CLASS_NAMES warningg (potential False positive)
}
```
In addition, SpotBugs does not contain any test cases for **EQ_COMPARING_CLASS_NAMES**. 
However, I found an example of the way SpotBugs will report **EQ_COMPARING_CLASS_NAMES** from [github](https://github.com/xvik/dropwizard-guicey/blob/33294c39521cf3fa2b0ac502ebbc97eb7f6c15e7/src/main/java/ru/vyarus/dropwizard/guice/module/lifecycle/UniqueGuiceyLifecycleListener.java).

I tried to write three more test cases to check the support level of Spotbug on **EQ_COMPARING_CLASS_NAMES**.
```java
public class EQ_COMPARING_CLASS_NAMES {
    /* From https://github.com/xvik/dropwizard-guicey/blob/33294c39521cf3fa2b0ac502ebbc97eb7f6c15e7/src/main/java/ru/vyarus/dropwizard/guice/module/lifecycle/UniqueGuiceyLifecycleListener.java */
    //  test01
    public boolean equals(final Object obj) {
        return getClass().getName().equals(obj.getClass().getName()); // waning.
    }

    // test02
    public boolean compare(final Object obj) {
        return getClass().getName().equals(obj.getClass().getName()); // no EQ_COMPARING_CLASS_NAMES warningg (potential False positive)
    }

    // test03
    public boolean test(final Object obj) {
        Object obj1 = new Object();
        if(obj.getClass().getName().equals(obj1.getClass().getName())){ // no EQ_COMPARING_CLASS_NAMES warningg (potential False positive)
            // do something
        }
    }
}
```
The test02 just replaces the method name 'equals' with 'compare'.
The test03 replaces the method name 'equals' with a casual name 'test' and adds an if branch. (common in practical projects). 
SpotBugs did not report **EQ_COMPARING_CLASS_NAMES** in **both** of them.

Does this mean that SpotBugs does not fully support **EQ_COMPARING_CLASS_NAMES**?
